packages/dataviews/src/dataforms-layouts/panel/modal.tsx

```diff
@@ -42,8 +46,40 @@ function ModalContent< Item >( {
 		setChanges( ( prev ) => ( { ...prev, ...value } ) );
 	};
 
-	// Merge original data with local changes for display
-	const displayData = { ...data, ...changes };
+	// Create flattened data and modified field definitions
+	const { flattenedData, modifiedFields } = useMemo( () => {
+		// It will be a mix of original structure for fields without getValue,
+		// and a flattened object in shape of { [fieldId]: fieldValue }
+		const newFlattenedData: { [ key: string ]: any } = {};
+
+		const newModifiedFields = fields.map( ( field ) => {
+			if ( field.getValue ) {
```

Comment:
should this check be tighter and check if it's a function too?
```suggestion
			if ( field.getValue && typeof field.getValue === 'function' ) {
```

Reply:
I would go in the opposite direction—I am not sure if we should gate it at all. In the fields normalization process, this function is always provided (even if not used by customers); see:

https://github.com/WordPress/gutenberg/blob/ddbcc73058a36d235813b3fa5b9704ea25475f49/packages/dataviews/src/normalize-fields.ts#L130

Reply:
Pushed that in https://github.com/WordPress/gutenberg/pull/71384/commits/f4957d22a4f48f2e02bd44fb949a918af2947e3c